com.unity.ml-agents/Runtime/SideChannels/IncomingMessage.cs

```diff
@@ -97,5 +112,14 @@ public void Dispose()
             m_Reader?.Dispose();
             m_Stream?.Dispose();
         }
+
+        /// <summary>
+        /// Whether or not there is more data left in the stream that can be read.
+        /// </summary>
+        /// <returns></returns>
+        bool CanReadMore()
```

Comment:
could this be inlined?

Reply:
[ [MethodImpl(MethodImplOptions.AggressiveInlining)]](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.methodimplattribute.-ctor?view=netframework-4.8#System_Runtime_CompilerServices_MethodImplAttribute__ctor)
```suggestion
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        bool CanReadMore()
```